pkg/api/controller.go

```diff
@@ -1731,6 +1731,134 @@ func (c *Controller) GetRepositoryMetadata(w http.ResponseWriter, r *http.Reques
 	writeResponse(w, r, http.StatusOK, apigen.RepositoryMetadata{AdditionalProperties: metadata})
 }
 
+func (c *Controller) GetBranchProtectionRules(w http.ResponseWriter, r *http.Request, repository string) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.GetBranchProtectionRulesAction,
+			Resource: permissions.RepoArn(repository),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	rules, eTag, err := c.Catalog.GetBranchProtectionRules(ctx, repository)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	resp := make([]*apigen.BranchProtectionRule, 0, len(rules.BranchPatternToBlockedActions))
+	for pattern := range rules.BranchPatternToBlockedActions {
+		resp = append(resp, &apigen.BranchProtectionRule{
+			Pattern: pattern,
+		})
+	}
+	if eTag == "" {
+		// workaround since swagger doesn't allow empty headers
+		// https://github.com/deepmap/oapi-codegen/issues/954
+		eTag = base64.StdEncoding.EncodeToString([]byte("EMPTY"))
+	}
```

Comment:
This is quite worrying.  Can we omit the etag if this happens?  What if we used openapi-codegen version 6.6.0, or 7.0.0?  @Jonathan-Rosenberg & I need to bump at least to 6.6.0 and probably to 7.0.0.

Reply:
Removed this.